codec.go

```diff
@@ -687,6 +642,66 @@ func appendStructFields(fields []structField, t reflect.Type, offset uintptr, se
 	return fields
 }
 
+func ambiguousNameTagCnt(names map[string]struct{}, embedded []embeddedField) (map[string]int, map[string]int) {
```

Comment:
I'm not sure what `Cnt` means in `ambiguousNameTagCnt`? Can you elaborate, or add docs to the function?

Reply:
Because `ambigousNames/ambigousTags` stores the count of the name/tag, so the `ambiguousNameTagCnt` means the count of name/tag.

How about to change `ambiguousNameTagCnt` to `ambiguousNamesAndTags`？
I think the self-explanatory function name is better.

Reply:
Ah, `Cnt` for `Count`. I obviously didn't have my morning coffee. I think just switching to `ambiguousNameTagCount` works 👍

Reply:
Ok, I fixed it.